src/backend/looped/weapons/aimbot.cpp

```diff
@@ -55,98 +56,99 @@ namespace big
-									local_fov_change = sqrt(pow(movement.x, 2) + pow(movement.y, 2));
-									mouse_movement.x = movement.x;
-									mouse_movement.y = movement.y;
-								}
-							}
-						}
 					}
 				}
 			}
-
-			if (PAD::GET_DISABLED_CONTROL_NORMAL(0, (int)ControllerInputs::INPUT_AIM))
+			if (!target_entity || ENTITY::IS_ENTITY_DEAD(target_entity, 0))
 			{
-				static bool update_time_now = true;
-				static std::chrono::system_clock::time_point current_time;
+				return;
+			}
+			if (PED::GET_PED_CONFIG_FLAG(self::ped, 78, 0))
+			{
+				Vector3 camera_target;
 
-				if (update_time_now)
+				if (g.weapons.aimbot.smoothing)
 				{
-					current_time    = std::chrono::system_clock::now();
-					update_time_now = false; //lockout
-				}
+					//Avoid buggy cam
+					if (using_aimbot_first_time)
+					{
+						Vector3 cam_coords               = CAM::GET_GAMEPLAY_CAM_COORD();
+						Vector3 cam_rot                  = CAM::GET_GAMEPLAY_CAM_ROT(0);
+						Vector3 cam_direction            = math::rotation_to_direction(cam_rot);
+						float distance                   = 150.f;
+						Vector3 multiply                 = cam_direction * distance;
+						Vector3 front_cam                = cam_coords + multiply;
+						camera_target                    = front_cam - CAM::GET_GAMEPLAY_CAM_COORD();
+						smooth_factor           = camera_target;
+						using_aimbot_first_time = false;
+					}
+					Vector3 target = aim_lock - CAM::GET_GAMEPLAY_CAM_COORD();
+					smooth_factor.x += (target.x - smooth_factor.x) * g.weapons.aimbot.smoothing_speed / 10.f;
+					smooth_factor.y += (target.y - smooth_factor.y) * g.weapons.aimbot.smoothing_speed / 10.f;
+					smooth_factor.z += (target.z - smooth_factor.z) * g.weapons.aimbot.smoothing_speed / 10.f;
 
-				std::chrono::duration<double> elapsed_time = std::chrono::system_clock::now() - current_time;
-				if (elapsed_time.count() > 0.f)
+					camera_target = smooth_factor;
+				}
+				else
 				{
-					INPUT mouse_handle; // MOUSEINPUT mi;
-					mouse_handle.type = INPUT_MOUSE;
-					mouse_handle.mi.dwFlags = MOUSEEVENTF_MOVE; // Type = Mouse movement, and the event  is emulating the mouse movement
-
-					// Update the mouse by moving it with how much we need / smoothing speed
-					mouse_handle.mi.dx = mouse_movement.x / (g.weapons.aimbot.smoothing ? g.weapons.aimbot.smoothing_speed : 2);
-					mouse_handle.mi.dy = mouse_movement.y / (g.weapons.aimbot.smoothing ? g.weapons.aimbot.smoothing_speed : 2);
-					SendInput(1, &mouse_handle, sizeof(mouse_handle)); //handles the input
-
-					//Reset our variables
-					mouse_movement.x = 0, mouse_movement.y = 0;
-					update_time_now = true; //reset our time
+					camera_target = aim_lock - CAM::GET_GAMEPLAY_CAM_COORD();
+				}
+				if (aim_lock.x == 0.f && aim_lock.y == 0.f && aim_lock.z == 0.f)
+					return;
+				//PEMDAS looks good
+
+				float camera_heading = atan2f(camera_target.x, camera_target.y) * 180.0f / std::numbers::pi;
```

Comment:
PEMDAS is not being followed here. You are letting the compiler decide the order of operations by not using parenthesis. So it will evaluate `atan2f(camera_target.x, camera_target.y)` first then it will multiply the result by 180 and then it will divide by Pi. You want what is essentially called RADPI, which is just 180 / pi. Define it in the header we discussed before, or at the top of this CPP file and make it a constexpr so the compiler knows it can evulate it.